src/relation_engine_server/utils/arango_client.py

```diff
@@ -108,6 +110,40 @@ def import_from_file(file_path, query):
     return resp.text
 
 
+def _init_readonly_user():
+    """
+    Using the root user, initialize an admin readonly user for use with ad-hoc queries.
+
+    If the user cannot be created, we raise an ArangoServerError
+    If the user already exists, or is successfully created, we return None and do not raise.
+    """
+    config = get_config()
+    user = config['db_readonly_user']
+    # Check if the user exists, in which case this is a no-op
+    resp = requests.get(
+        config['db_url'] + '/_api/user/' + user,
+        auth=(config['db_user'], config['db_pass'])
+    )
+    if resp.status_code == 200:
+        return
+    # Create the user
+    resp = requests.post(
+        config['db_url'] + '/_api/user',
+        data=json.dumps({'user': user, 'passwd': config['db_readonly_user']}),
+        auth=(config['db_user'], config['db_pass'])
+    )
```

Comment:
I like that this makes deployment easy as the user does not need to already exist. On the other hand, I assume this requires root access which requires the service to have the root privileges. I had thought we might want more limited privs for the api user like write access to the particular environment databases

Reply:
We could do that. db_user and db_pass can be set to a non-root user with 'rw' access to the specific database. That user will be able to create and grant 'ro' access for another user on the same database. Does that sound good? Doesn't require code changes

Reply:
That's good. From the docs I didn't find it clear that you can create a new user w/o admin on _system.